src/v0/destinations/google_adwords_offline_conversions/utils.js

```diff
@@ -362,6 +363,55 @@ const getClickConversionPayloadAndEndpoint = (message, Config, filteredCustomerI
   return { payload, endpoint };
 };
 
+const GOOGLE_ALLOWED_CONSENT_STATUS = ['UNSPECIFIED', 'UNKNOWN', 'GRANTED', 'DENIED'];
+
+/**
+ * Populates the consent object based on the provided properties.
+ *
+ * @param {object} properties - message.properties containing properties related to consent.
+ * @returns {object} - An object containing consent information.
+ *  * ref :
+ * 1) For click conversion :
+ *  a) https://developers.google.com/google-ads/api/rest/reference/rest/v15/customers/uploadClickConversions#ClickConversion
+ *  b) https://developers.google.com/google-ads/api/reference/rpc/v15/ClickConversion#consent
+ * 2) For Call conversion :
+ *  a) https://developers.google.com/google-ads/api/rest/reference/rest/v15/customers/uploadCallConversions#CallConversion
+ *  b) https://developers.google.com/google-ads/api/reference/rpc/v15/CallConversion#consent
+ * 3) For Store sales conversion :
+ *  a) https://developers.google.com/google-ads/api/reference/rpc/v15/UserData
+ *  b) https://developers.google.com/google-ads/api/reference/rpc/v15/UserData#consent
+ */
+
+const populateConsentForGoogleDestinations = (message) => {
```

Comment:
We need to make this a generic utility and move it out.

Let's take in inputs like consent type fields and destination name.

Reply:
had done differently